### PR TITLE
[Security] remove deprecated HTTP digest auth key 

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Security/Http/Firewall/DigestAuthenticationListener.php [hunk 1]
```diff
@@ -78,7 +78,7 @@ public function handle(GetResponseEvent $event)
         }
 
         try {
-            $digestAuth->validateAndDecode($this->authenticationEntryPoint->getKey(), $this->authenticationEntryPoint->getRealmName());
+            $digestAuth->validateAndDecode($this->authenticationEntryPoint->getSecret(), $this->authenticationEntryPoint->getRealmName());
```
Comment (xabbuh) [inline src/Symfony/Component/Security/Http/Firewall/DigestAuthenticationListener.php]: We do not seem to have tests covering this. But they would need to be added in older versions.

**Comment by WouterJ** [comment]
:+1:  Status: reviewed

**Comment by xabbuh** [comment]
Looks like we should remove the config option for this in the SecurityBundle too.

File: src/Symfony/Component/Security/Http/Firewall/DigestAuthenticationListener.php (hunk 1 above)
Comment (fabpot) [inline src/Symfony/Component/Security/Http/Firewall/DigestAuthenticationListener.php]: and we should make this change on 2.8 as well, right?

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**